x-pack/plugins/security_solution/public/overview/components/overview_empty/index.tsx

```diff
@@ -6,105 +6,57 @@
  */
 
 import React, { useMemo } from 'react';
-import { omit } from 'lodash/fp';
-import { createStructuredSelector } from 'reselect';
-
-import { FormattedMessage } from '@kbn/i18n/react';
-import { EuiLink } from '@elastic/eui';
-import * as i18nCommon from '../../../common/translations';
-import { EmptyPage, EmptyPageActionsProps } from '../../../common/components/empty_page';
+import { i18n } from '@kbn/i18n';
 import { useKibana } from '../../../common/lib/kibana';
 import { ADD_DATA_PATH } from '../../../../common/constants';
-import {
-  useEndpointSelector,
-  useIngestUrl,
-} from '../../../management/pages/endpoint_hosts/view/hooks';
-import { useNavigateToAppEventHandler } from '../../../common/hooks/endpoint/use_navigate_to_app_event_handler';
-import { CreateStructuredSelector } from '../../../common/store';
-import { endpointPackageVersion as useEndpointPackageVersion } from '../../../management/pages/endpoint_hosts/store/selectors';
+import { pagePathGetters } from '../../../../../fleet/public';
+import { SOLUTION_NAME } from '../../../../public/common/translations';
 import { useUserPrivileges } from '../../../common/components/user_privileges';
 
+import {
+  KibanaPageTemplate,
+  NoDataPageActionsProps,
+} from '../../../../../../../src/plugins/kibana_react/public';
+
 const OverviewEmptyComponent: React.FC = () => {
   const { http, docLinks } = useKibana().services;
   const basePath = http.basePath.get();
-  const selector = (createStructuredSelector as CreateStructuredSelector)({
-    endpointPackageVersion: useEndpointPackageVersion,
-  });
-  const { endpointPackageVersion } = useEndpointSelector(selector);
-  const { url: ingestUrl } = useIngestUrl('');
-
-  const endpointIntegrationUrlPath = endpointPackageVersion
-    ? `/endpoint-${endpointPackageVersion}/add-integration`
-    : '';
-  const endpointIntegrationUrl = `/integrations${endpointIntegrationUrlPath}`;
-  const handleEndpointClick = useNavigateToAppEventHandler('fleet', {
-    path: endpointIntegrationUrl,
-  });
   const canAccessFleet = useUserPrivileges().endpointPrivileges.canAccessFleet;
+  const integrationsPathComponents = pagePathGetters.integrations_all({ category: 'security' });
```

Comment:
@jen-huang 

I'm using this to get the integrations + category link for use with this "Add Data" screen.  The URL we're picking up is `/app/integrations/browse/security`

This navigates us to the Integrations page, however, it doesn't seem the `security` category is selected by default.

The result I see is that it navigates to the integrations page, but it drops the `security` category from the URL and defaults to the All category.

![image](https://user-images.githubusercontent.com/56395104/134404061-ff513a75-63ae-4ed8-9441-f04894288e17.png)

If I selected Security, I do see `security` get appended to the URL, but if I refresh the page with this URL, it's still not persisted.

Select "Security" -
![image](https://user-images.githubusercontent.com/56395104/134404153-27eabd3b-5cc0-4727-aefb-e54efcd17435.png)

After refresh, Security is dropped:
![image](https://user-images.githubusercontent.com/56395104/134404212-1aca31ee-e1af-4ff9-bc2e-7a26191bdc71.png)

Is there a bug here?  Should I be able to navigate to the Integrations page with a Category selected?

Reply:
this is almost certainly a bug, you should definitely be able to browse to a specific category given the URL path. could you file a separate issue? thanks in advance!

Reply:
Thanks @jen-huang - I filed the above in this ticket: https://github.com/elastic/kibana/issues/112871

@cchaos note on the above - as it stands, this navigation won't sort the integrations by Security until the above bug is fixed.  The navigation still works, but the user won't see the Security integrations sorted.  We could merge this as is and it should work after the bug is fixed.  Thoughts?

Reply:
Sounds good. Let's not block this PR because of the routing bug.

Reply:
FYI this bug was fixed so this should work now. let me know if you still encounter issues